File: online-judge-solutions/Codeforces/454A.cpp
//https://codeforces.com/problemset/problem/454/A
//Lang		:	GNU C++14
//Time		:	15 ms
//Memory	:	0 KB
#include<iostream>
using namespace std;
int main(){
	int n;cin>>n;
	for(int i=0;i<n;i++){
		string s(n,'*');
		int c=i;
		if(c>n/2) c=n-1-c;
		c=2*c+1;
		for(int j=(n-c)/2;j<((n-c)/2)+c;j++) s[j]='D';
		cout<<s<<endl;
	}
	return 0;
}

File: online-judge-solutions/Codeforces/454A.py
#https://codeforces.com/problemset/problem/454/A
#Lang		:	Python 3.8
#Time		:	46 ms
#Memory         :	0 KB
n=int(input())
for i in range(n):
    s,c='*'*n,i
    if(c>int(n/2)): c=n-1-c
    c=int(2*c+1)
    for j in range(int((n-c)/2)+1,int((n-c)/2)+c+1,1):s=s[:j-1]+'D'+s[j:]
    print(s)
